compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangParserListener.java

```diff
@@ -3038,16 +3042,65 @@ private String getNodeValue(ParserRuleContext ctx, TerminalNode node) {
+                this.pkgBuilder.addLiteralValue(pos, ws, TypeTags.INT, Long.MAX_VALUE);
+                if (nodeValue.startsWith("-")) {
+                    dlog.error(pos, DiagnosticCode.INTEGER_TOO_SMALL, nodeValue);
+                } else {
+                    dlog.error(pos, DiagnosticCode.INTEGER_TOO_LARGE, nodeValue);
+                }
+            }
         } else if (integerLiteralContext.HexIntegerLiteral() != null) {
-            return Long.parseLong(getNodeValue(simpleLiteralContext, integerLiteralContext.HexIntegerLiteral())
-                    .toLowerCase().replace("0x", ""), 16);
+            String nodeValue = getNodeValue(simpleLiteralContext, integerLiteralContext.HexIntegerLiteral());
+            try {
+                return Long.parseLong(nodeValue.toLowerCase().replace("0x", ""), 16);
+            } catch (Exception e) {
+                DiagnosticPos pos = getCurrentPos(simpleLiteralContext);
+                Set<Whitespace> ws = getWS(simpleLiteralContext);
+                // Assign a value and continue the compilation. Since there is an error, program will not run.
+                this.pkgBuilder.addLiteralValue(pos, ws, TypeTags.INT, Long.MAX_VALUE);
+                if (nodeValue.startsWith("-")) {
+                    dlog.error(pos, DiagnosticCode.HEXADECIMAL_TOO_SMALL, nodeValue);
+                } else {
+                    dlog.error(pos, DiagnosticCode.HEXADECIMAL_TOO_LARGE, nodeValue);
+                }
+            }
         } else if (integerLiteralContext.OctalIntegerLiteral() != null) {
-            return Long.parseLong(getNodeValue(simpleLiteralContext, integerLiteralContext.OctalIntegerLiteral())
-                    .replace("0_", ""), 8);
+            String nodeValue = getNodeValue(simpleLiteralContext, integerLiteralContext.OctalIntegerLiteral());
+            try {
+                return Long.parseLong(nodeValue.replace("0_", ""), 8);
+            } catch (Exception e) {
+                DiagnosticPos pos = getCurrentPos(simpleLiteralContext);
+                Set<Whitespace> ws = getWS(simpleLiteralContext);
+                // Assign a value and continue the compilation. Since there is an error, program will not run.
+                this.pkgBuilder.addLiteralValue(pos, ws, TypeTags.INT, Long.MAX_VALUE);
+                if (nodeValue.startsWith("-")) {
+                    dlog.error(pos, DiagnosticCode.OCTAL_TOO_SMALL, nodeValue);
+                } else {
+                    dlog.error(pos, DiagnosticCode.OCTAL_TOO_LARGE, nodeValue);
+                }
+            }
         } else if (integerLiteralContext.BinaryIntegerLiteral() != null) {
-            return Long.parseLong(getNodeValue(simpleLiteralContext, integerLiteralContext.BinaryIntegerLiteral())
-                    .toLowerCase().replace("0b", ""), 2);
+            String nodeValue = getNodeValue(simpleLiteralContext, integerLiteralContext.BinaryIntegerLiteral());
```

Comment:
This code block seems similar for all four cases. Shall we refactor this and bring to a common method, and reuse that method inside the four if-statements?